scripts/scil_compute_local_tracking.py

```diff
@@ -164,13 +207,114 @@ def tqdm_if_verbose(generator: Iterable, verbose: bool, *args, **kwargs):
+                                affine_to_rasmm=seed_img.affine)
+
+    filetype = nib.streamlines.detect_format(args.out_tractogram)
+    reference = get_reference_info(seed_img)
+    header = create_tractogram_header(filetype, *reference)
+
+    # Use generator to save the streamlines on-the-fly
+    nib.streamlines.save(tractogram, args.out_tractogram, header=header)
+
+
+def save_tractogram_gpu(streamlines_generator, odf_sh_img, args):
+    voxel_size = odf_sh_img.header.get_zooms()[0]
+
+    # wrapper for tracker.track() yielding one TractogramItem per
+    # streamline for use with the LazyTractogram.
+    def tracks_generator_wrapper():
+        for strl, seed in streamlines_generator:
+            # seeds are returned with origin `center`
+            dps = {}
+            if args.save_seeds:
+                dps['seeds'] = seed
+
+            # IMPORTANT: Streamlines are dumped in world space (mm).
+            # TODO: Investigate why the streamline must NOT be shifted to
+            # origin `center` for LazyTractogram.
+            strl *= voxel_size  # in mm.
+            if args.compress:
+                strl = compress_streamlines(strl, args.compress)
+            yield TractogramItem(strl, dps, {})
+
+    # instantiate tractogram
+    tractogram = LazyTractogram.from_data_func(tracks_generator_wrapper)
+    tractogram.affine_to_rasmm = odf_sh_img.affine
+
+    filetype = nib.streamlines.detect_format(args.out_tractogram)
+    reference = get_reference_info(odf_sh_img)
+    header = create_tractogram_header(filetype, *reference)
+
+    # Use generator to save the streamlines on-the-fly
+    nib.streamlines.save(tractogram, args.out_tractogram, header=header)
+
+
 def main():
+    t_init = perf_counter()
     parser = _build_arg_parser()
     args = parser.parse_args()
 
+    # TODO: Disable logging from pyopencl library.
     if args.verbose:
         logging.getLogger().setLevel(logging.DEBUG)
 
+    if args.use_gpu:
+        batch_size = args.batch_size or DEFAULT_BATCH_SIZE
+        sh_interp = args.sh_interp or DEFAULT_SH_INTERP
+        forward_only = args.forward_only or DEFAULT_FWD_ONLY
+        if args.algo != 'prob':
+            parser.error('Algo `{}` not supported for GPU tracking. '
+                         'Set --algo to `prob` for GPU tracking.'
+                         .format(args.algo))
+        if args.sphere != DEFAULT_GPU_SPHERE:
+            parser.error('Cannot use sphere `{}`. Only symmetric724 is '
+                         'available for GPU tracking.'.format(args.sphere))
+        if args.sub_sphere:
+            parser.error('Invalid argument --sub_sphere. Not implemented '
+                         'for GPU.')
+    else:
+        if args.batch_size is not None:
+            parser.error('Invalid argument --batch_size. '
+                         'Set --use_gpu to enable.')
+        if args.sh_interp is not None:
+            parser.error('Invalid argument --sh_interp. '
```

Comment:
Similarly to my comment above, the warning here and next line should not be that it is invalid. It should be that it is only available with GPU. It could even be a NotImplementedError if you wish!